src/data/error.rs

```diff
@@ -287,6 +324,45 @@ pub enum LexError {
     #[error("no newline at end of file")]
     NoNewlineAtEOF,
 
+    #[error("unknown token: {0:?}")]
+    UnknownToken(u8),
+
+    #[error("missing terminating \" character in string literal")]
+    StringMissingEndQuote,
+
+    #[error("illegal newline while parsing string literal")]
+    NewlineInString,
+
+    #[error("{0} character escape out of range")]
+    InvalidNumericCharEscape(Radix),
```

Comment:
Maybe `CharEscapeOutOfRange` would be better? `InvalidNumeric` doesn't tell you why it was better. `OutOfRange` seems a little verbose though ... I'm open to suggestions.

Reply:
I would have chosen `InvalidCharEscape`, but that makes it seem like it might be an incorrect letter after a slash in a char literal (ex: `\j`)

Reply:
Right, my point is that for `InvalidNumericCharEscape` to make sense, you have to already know the only thing that can make it invalid is an escape that's out of range.

Reply:
Okay, I will just rename it to `CharEscapeOutOfRange` then.